File: C_Little_Girl_and_Maximum_Sum.cpp
#include <bits/stdc++.h>
using namespace std;

#define int long long

signed main() {
    int n, q, ans = 0;
    cin >> n >> q;
    vector<int> a(n), diff(n + 2), cnt;
    for (int &ai : a) {
        cin >> ai;
    }
    while (q--) {
        int l, r;
        cin >> l >> r;
        diff[l]++;
        diff[r + 1]--;
    }
    for (int i = 1; i <= n; ++i) {
        diff[i] += diff[i - 1];
        cnt.emplace_back(diff[i]);
    }
    ranges::sort(cnt, greater());
    ranges::sort(a, greater());
    for (int i = 0; i < n; ++i) {
        ans += cnt[i] * a[i];
    }
    cout << ans << endl;
    return 0;
}

File: C_Little_Girl_and_Maximum_Sum.py
n, q = map(int, input().split())
a = list(map(int, input().split()))
diff, cnt = [0 for _ in range(n + 2)], []
for _ in range(q):
    l, r = map(int, input().split())
    diff[l] += 1
    diff[r + 1] -= 1
for i in range(1, n + 1):
    diff[i] += diff[i - 1]
    cnt.append(diff[i])
a.sort(reverse = True)
cnt.sort(reverse = True)
print(sum(cnt[i] * a[i] for i in range(n)))
